src/main/java/org/reaktivity/k3po/nukleus/ext/internal/behavior/NukleusSource.java

```diff
@@ -39,7 +42,7 @@
     private final String sourceName;
     private final MutableDirectBuffer writeBuffer;
 
-    private final Long2ObjectHashMap<NukleusServerChannel> routesByRef;
+    private final Map<List<Long>, NukleusServerChannel> routesByRef;
```

Comment:
Store `authorization` on `NukleusServerChannel`?

Reply:
I want to support multiple routes with same sourceRef, just as we need to do in nuklei. But thinking about it some more that might not be needed, since k3po is always the endpoint, unlike nuklei (this is similar to why we need exact match on authorization for accept).

Reply:
Is it possible to do this without needing to construct a new object for each map lookup?

Reply:
I could have a multi-level structure (value is array of NukleusServerChannel) but it seems more complex. I would certainly have done it for reaktor runtime code but this is testware so not performance critical.

Reply:
done